single_kernel_mongo/state/charm_state.py

```diff
@@ -118,7 +116,14 @@ def peers_units(self) -> set[Unit]:
 
     @property
     def client_relations(self) -> set[Relation]:
-        """The set of client relations."""
+        """The set of client relations.
+
+        Client relations exist on two separate interfaces, one for sharding,
+        which is exposed for mongos charms, and one for replication which is
+        exposed for mongodb charms.
+        """
+        if self.charm_role.name == KindEnum.MONGOS:
```

Comment:
I believe we should find a better name for this Enum and getting rid of the `Enum` suffix.

Reply:
You're probably right.
I'm trying to think about something that is not "Role" because role means replication vs shard vs config-server vs mongos.
Maybe just `Kind`?